AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -71,8 +72,44 @@ def create_new_variable(self, new_variable_type: str) -> None:
         table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
         variable_entry = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
 
-        # Click on it to create variable
+        # todo
+        # maybe put these 3 steps into a helper function
+        # maybe assert here ?
+
+        # Click on it to create variable and wait 1 second for the UI to respond
         pyside_utils.item_view_index_mouse_click(table_view, variable_entry)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
+
+        # Set the new variable's name. Wait 1 second for the UI to respond
+        graph_vars_page = self.variable_manager.findChild(QtWidgets.QWidget, GRAPH_VARIABLES_PAGE_QT)
+        line_edit = graph_vars_page.findChild(QtWidgets.QLineEdit, "")
+        line_edit.insert(variable_name)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
+
+        # Store our selection so we can return it. Clear the selection and wait 1 second for the UI to respond
+        selection_model = graph_vars_page.findChild(QtCore.QItemSelectionModel)
+        selection = selection_model.selection()
+        selection_model.clear()
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
```

Comment:
These comments tell me that there's helper functions to be extracted here.

Reply:
broke this down into a sequence of helper functions